aeron-archive/src/main/java/io/aeron/archive/ReplicationSession.java

```diff
@@ -516,7 +531,7 @@ private int extend()
 
         if (null == recordingSubscription)
         {
-            state(State.DONE);
+            state(State.DONE, "null recordingSubscription");
```

Comment:
The `recordingSubscription` will be `null` only if `conductor.extendRecording` call fails with an error. The error message is directly sent via the ControlSession back to the AeronArchive client. The response will be logged (if enabled) via `CMD_OUT_RESPONSE` event. However it would be great if we could get the error message and capture it in a `reason` field here. That would require changing `conductor.extendRecording` method (e.g. return `Object` instead of `Subscription` and return error String if error and actual subscription if success).